language/tools/move-package/tests/test_runner.rs

```diff
@@ -2,135 +2,164 @@
-                    generate_abis: false,
-                    install_dir: Some(tempdir().unwrap().path().to_path_buf()),
-                    force_recompilation: false,
-                    ..Default::default()
-                },
-                &mut Vec::new(), /* empty writer as no diags needed */
+
+    let expected = fs::read_to_string(&test.expected)?;
+    if expected != output {
+        return Err(anyhow!(add_update_baseline_fix(format!(
+            "Expected outputs differ for {:?}:\n{}",
+            test.expected,
+            format_diff(expected, output),
+        )))
+        .into());
+    }
+
+    Ok(())
+}
+
+struct Test<'a> {
+    toml_path: &'a Path,
+    expected: PathBuf,
+    output_dir: TempDir,
+}
+
+impl Test<'_> {
+    fn from_path(toml_path: &Path) -> datatest_stable::Result<Test> {
+        let mut candidates = EXTENSIONS
+            .iter()
+            .map(|ext| toml_path.with_extension(*ext))
+            .filter(|p| p.is_file());
+
+        let Some(expected) = candidates.next() else {
+            return Err(anyhow!(
+                "No snapshot file found for {:?}, please add a file with the same basename and one \
+                 of the following extensions: {:#?}\n\n\
+                 You probably want to re-run with `env UPDATE_BASELINE=1` after adding this file.",
+                toml_path,
+                EXTENSIONS,
+            ).into());
+        };
+
+        let mut remaining: Vec<_> = candidates.collect();
+        if !remaining.is_empty() {
+            remaining.push(expected);
+            return Err(anyhow!(
+                "Multiple snapshot files found for {:?}, please supply only one of: {:#?}",
+                toml_path,
+                remaining,
             )
+            .into());
+        }
+
+        Ok(Test {
+            toml_path,
+            expected,
+            output_dir: tempdir()?,
         })
-        .and_then(|rg| rg.resolve())
-    {
-        Ok(mut resolved_package) => match (should_compile, should_model) {
-            (true, true) => {
-                return Err(anyhow::format_err!(
-                    "Cannot have compile and model flags set for same package"
+    }
+
+    fn run(&self) -> anyhow::Result<String> {
+        let config = BuildConfig {
+            dev_mode: true,
+            test_mode: false,
+            generate_docs: false,
+            generate_abis: false,
+            install_dir: Some(self.output_dir.path().to_path_buf()),
+            force_recompilation: false,
+            ..Default::default()
+        };
+
+        let mut sink = std::io::sink();
+        let mut resolved_package =
+            config.resolution_graph_for_package(self.toml_path, &mut sink)?;
+
+        let Some(ext) = self.expected.extension().and_then(OsStr::to_str) else {
+            bail!("Unexpected snapshot file extension: {:?}", self.expected.extension());
+        };
+
+        Ok(match ext {
+            "compiled" => {
```

Comment:
anyway to bring back the constants for these?

Reply:
It's possible (to match against a `const`), but I was finding that pulling these values out into constants wasn't really helping me, because the constant names matched the contents of the string literal, so if I changed one, I would probably change the other anyway, and they are only used in two places (here and in the list of all extensions), so the extra indirection wasn't providing me with much.

That being said, I don't feel strongly about this, so let me know if you feel slightly stronger, and I can pull these out into constants.

Reply:
No strong feelings, just a knee-jerk reaction when I see important values like this sitting all alone without a cozy `const` to keep them safe